### PR TITLE
Update ci-main.yml

### PR DESCRIPTION
**Pull request by cooljeanius**
bump macOS version; make use of all available processors

### CODE REVIEW

**Comment by cooljeanius** [comment]
Failures now look like:
```
WARNING: Client prerequisites are not met. wesnoth cannot be built.
WARNING: Unit tests are disabled because their prerequisites are not met
  If any config checks fail, look in build/config.log for details
  If a check fails spuriously due to caching, use --config=force to force its rerun
---[applying configuration]---
SConscript:/Users/runner/work/wesnoth/wesnoth/po/SConscript  took 220.983 ms
SConscript:/Users/runner/work/wesnoth/wesnoth/doc/manual/SConscript  took 0.225 ms
SConscript:/Users/runner/work/wesnoth/wesnoth/doc/SConscript  took 0.514 ms
SConscript:/Users/runner/work/wesnoth/wesnoth/packaging/windows/SConscript  took 0.484 ms
SConscript:/Users/runner/work/wesnoth/wesnoth/packaging/systemd/SConscript  took 0.349 ms
SConscript:/Users/runner/work/wesnoth/wesnoth/build/release/SConscript  took 392.092 ms

scons: *** Export of non-existent variable ''wesnoth''
SConscript:/Users/runner/work/wesnoth/wesnoth/SConstruct  took 10693.099 ms
File "/Users/runner/work/wesnoth/wesnoth/src/SConscript", line 159, in WesnothProgram
Error: Process completed with exit code 2.
```
@CelticMinstrel @Vultraz any ideas?

**Comment by loonycyborg** [comment]
From which job this output? "scons translations" on macos?

**Comment by loonycyborg** [comment]
Maybe try it with 2b965290f60df5f7f677da388af308c7aca8d333 reverted.

**Comment by cooljeanius** [comment]
> Maybe try it with [2b96529](https://github.com/cooljeanius/wesnoth/commit/2b965290f60df5f7f677da388af308c7aca8d333) reverted.

Huh, it looks like it already has been? I guess I'll try the other way around...

File: src/SConscript
Comment (cooljeanius) [inline src/SConscript:1]: (I couldn't find the "cherry-pick" button, so this is a manual copying of 2b96529 to this branch)

**Comment by cooljeanius** [comment]
...ok, so now the failure is:
```
/Users/runner/work/wesnoth/wesnoth/projectfiles/Xcode/Headers/boost/container_hash/hash.hpp:132:33: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
        struct hash_base : std::unary_function<T, std::size_t> {};
                           ~~~~~^
```

**Comment by loonycyborg** [comment]
> > Maybe try it with [2b96529](https://github.com/cooljeanius/wesnoth/commit/2b965290f60df5f7f677da388af308c7aca8d333) reverted.
> 
> Huh, it looks like it already has been? I guess I'll try the other way around...

well this change definitely exists in current master. This PR is targeting your own personal fork master branch and not upstream master, and it's quite old. But it's good that this SConscript change fixes the problem rather than causes.

**Comment by cooljeanius** [comment]
> > > Maybe try it with [2b96529](https://github.com/cooljeanius/wesnoth/commit/2b965290f60df5f7f677da388af308c7aca8d333) reverted.
> > 
> > 
> > Huh, it looks like it already has been? I guess I'll try the other way around...
> 
> well this change definitely exists in current master. This PR is targeting your own personal fork master branch and not upstream master, and it's quite old.

Yes, I hope to eventually get back in sync with upstream, but it's quite difficult to do so, as can be seen by the number of merge conflicts in #148, so I'm reduced to attempting to cherry-pick certain commits for now, until I figure things out...

**Review by cooljeanius — COMMENTED** [review]
OK

**Pull request merged**